### PR TITLE
Fix issues in differential FVA

### PR DESCRIPTION
**Pull request by Midnighter**
A number of issues surfaced with differential FVA:

1. Correct normalization of flux ranges
2. Correct the conditions for suddenly essential reactions
3. Exclude cases where the envelope is computed based on more than one variable

Please note that two test cases fail because now the reference files are, of course, incorrect. @phantomas1234 were those references generated with Mathematica?

### CODE REVIEW

**Comment by phantomas1234** [comment]
@Midnighter  I don't think the test data came from the Mathematica implementation. Can you run the PR version of DifferentialFVA on the serine production case and then we compare with the Mathematica notebook to be sure it is correct now?

**Comment by phantomas1234** [comment]
Also why is there no travis-ci build on this?

**Comment by Midnighter** [comment]
The serine case is running. We can look at results tomorrow.

> Also why is there no travis-ci build on this?

Since this PR is targetting the other branch, I think the current Travis config disables it.

File: cameo/strain_design/deterministic/flux_variability_based.py [hunk 1]
```diff
@@ -369,8 +440,8 @@ def run(self, surface_only=True, improvements_only=True, progress=True, view=Non
             ] = True
 
             df.loc[
-                ((df.lower_bound <= 0) & (df.lower_bound > 0)) | (
-                    (ref_lower_bound >= 0) & (df.upper_bound <= 0)),
+                (zero_overlap_mask & (df.lower_bound > 0)) | (
```
Comment (phantomas1234) [inline cameo/strain_design/deterministic/flux_variability_based.py]: How reliable is this in terms of numerical issues? Should this maybe use `non_zero_flux_threshold` or similar?

File: cameo/strain_design/deterministic/flux_variability_based.py [hunk 2]
```diff
@@ -344,6 +410,11 @@ def run(self, surface_only=True, improvements_only=True, progress=True, view=Non
         ref_lower_bound = self.reference_flux_ranges.lower_bound.apply(
             lambda v: 0 if abs(v) < non_zero_flux_threshold else v)
```
Comment (phantomas1234) [inline cameo/strain_design/deterministic/flux_variability_based.py]: is `ref_lower_bound` obsolete now? Maybe it should be used for `zero_overlap_mask` below?

File: cameo/strain_design/deterministic/flux_variability_based.py (hunk 1 above)
Comment (phantomas1234) [reply cameo/strain_design/deterministic/flux_variability_based.py]: I mean something like `zero_overlap_mask & (df.lower_bound > non_zero_flux_threshold)`

Comment (Midnighter) [reply cameo/strain_design/deterministic/flux_variability_based.py]: What about modifying all solutions as in the following:
```py
df[df[["lower_bound", "upper_bound"]].abs() < non_zero_flux_threshold] = 0
```

File: cameo/strain_design/deterministic/flux_variability_based.py (hunk 2 above)
Comment (Midnighter) [reply cameo/strain_design/deterministic/flux_variability_based.py]: Right, I didn't take care of this.

File: cameo/strain_design/deterministic/flux_variability_based.py (hunk 1 above)
Comment (phantomas1234) [reply cameo/strain_design/deterministic/flux_variability_based.py]: fine with me

**Comment by Midnighter** [comment]
Alright, coverage decreases a bit but overall it's an improvement.

**Pull request merged**